### ISSUE TITLE
[Question] Performance improvement for git_prompt_vars

### ISSUE THREAD

**Issue by sharils** [Bash-it/bash-it#1244]
The `git_prompt_vars` function could take up to 500ms to run on Mac because it needs to run tons of git commands to aggregate all info needed.

And I just found that one single `git status --show-stash` command contains all the necessary information (detached or not, which commit if detached, which branch, whether there is a remote branch, the name of the remote branch, count of commits ahead or behind the remote branch, number of stashed entries) and it only takes about 20ms.

But the problem is that the format is hard to parse and the ``--porcelain`` switch doesn't show those info and of course the parser might be broken in the future version of git.

So my question is: Would it be a good idea to send a PR that replaces the current `git_prompt_vars` implementation by parsing the result from a single `git status --show-stash` command?

Thanks.

**Comment by nwinkler** [comment]
Sounds like a good improvement, but comes with the downsides that you mention. Based on that, I'm not really in favor of putting something in Bash-it that will likely become a maintenance issue in the future.

To be honest, the right thing to do would be fixing this in Git itself: Either by requesting that the `--porcelain` option should contain the missing info, or by asking the Git team to possibly speed up the expensive stash call. On the other hand, that might exactly be the reason why this info is not currently being included...

**Comment by sharils** [comment]
That makes sense. Let's close this for now. I'll see what I can do to info the Git team.
